plugins/modules/zos_zfs_resize.py

```diff
@@ -0,0 +1,660 @@
+
+- name: Resize an aggregate data set and get the full trace on an uss file and verbose output.
+  zos_zfs_resize:
+    target: TEST.ZFS.DATA
+    size: 2500
+    verbose: True
+    trace_destination: /tmp/helper.txt
+"""
+
+RETURN = r"""
+cmd:
+    description: The zfsadm command executed on the remote node.
+    returned: always
+    type: str
+    sample: zfsadm grow -aggregate SOMEUSER.VVV.ZFS -size 4096
+target:
+    description: The Fully Qualified Name of the resized zfs data set.
+    returned: always
+    type: str
+    sample: SOMEUSER.VVV.ZFS
+mount_target:
+    description: The original share/mount.
+    returned: always
+    type: str
+    sample: /tmp/zfs_agg
+size:
+    description: The approximate size, of the data set after the resizing is performed.
+    returned: always
+    type: int
+    sample: 4024
+rc:
+    description: The return code of the zfsadm command.
+    returned: always
+    type: int
+    sample: 0
+old_size:
+    description: The reported size, in space_type, of the data set before the resizing is performed.
+    returned: always
+    type: float
+    sample: 3096
+old_free_space:
+    description: The reported size, in space_type, of the free space in the data set before the resizing is performed.
+    returned: always
+    type: float
+    sample: 108
+new_size:
+    description: The reported size, in space_type, of the data set after the resizing is performed.
+    returned: success
+    type: float
+    sample: 4032
+new_free_space:
+    description: The reported size, in space_type, of the free space in the data set after the resizing is performed.
+    returned: success
+    type: float
+    sample: 48
+space_type:
+    description: The reported space_type that the size and free variables are reported.
+    returned: always
+    type: str
+    sample: k
+verbose_output:
+    description: If C(verbose=true) the full traceback of operation will show on this variable. If C(trace) will return the data set or path name.
+    returned: C(verbose=true) and success
+    type: str
+    sample: 6FB2F8 print_trace_table printing contents of table Main Trace Table
+"""
+
+import os
+import tempfile
+from pathlib import Path
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.ibm.ibm_zos_core.plugins.module_utils import (
+    better_arg_parser,
+    data_set,
+)
+
+from ansible_collections.ibm.ibm_zos_core.plugins.module_utils.zfsadm import zfsadm
+
+
+def calculate_size_on_k(size, space_type):
+    """Function to convert size depending on the space_type.
+
+    Parameters
+    ----------
+        size : int
+            Size to grow or shrink the zfs
+        space_type : str
+            Type of space to be use
+
+    Returns
+    -------
+        size : int
+            Size on kilobytes
+    """
+    if space_type == "m":
+        size *= 1024
+    if space_type == "g":
+        size *= 1048576
+    if space_type == "cyl":
+        size *= 830
```

Comment:
Could this equivalence be wrong for certain kinds of volumes? Or are we sure that a cylinder will always be 830KB?

Reply:
Here is the [documentation](https://www.ibm.com/docs/en/zoau/1.3.x?topic=commands-dtouch#unit-conversions)
<img width="759" alt="Captura de pantalla 2024-12-26 a la(s) 2 10 46 p m" src="https://github.com/user-attachments/assets/04f4599d-8e47-4ba0-a37b-03bd65123313" />